pydantic/tools.py

```diff
@@ -39,7 +22,7 @@ def parse_obj_as(type_: type[T], obj: Any, type_name: NameFactory | None = None)
         warnings.warn(
             'The type_name parameter is deprecated. parse_obj_as no longer creates temporary models', stacklevel=2
         )
-    return Validator(type_)(obj)
+    return funcs.analyze_type(type_).validate_python(obj)
```

Comment:
We should probably explicitly deprecate `parse_obj_as` if this is the intended path now. Don't think we need to do that prior to alpha though. Maybe adding a TODO comment about it is a good idea though.

Reply:
I agree we should deprecate it.